apps/sms/views/shared/test/unit/navigation_test.js

```diff
@@ -166,6 +169,34 @@ suite('navigation >', function() {
           sinon.assert.calledTwice(console.error);
         }).then(done, done);
       });
+
+      test('Set init pending when launched by notification', function(done) {
+        this.sinon.stub(console, 'error');
+        navigator.mozHasPendingMessage.withArgs('notification').returns(true);
+
+        assert.isFalse(Navigation.hasPendingInit());
+        Navigation.init().catch(() => {}).then(() => {
+          assert.isTrue(Navigation.hasPendingInit());
+          sinon.assert.notCalled(fakeWindow.InboxView.beforeEnter);
+          sinon.assert.notCalled(fakeWindow.InboxView.afterEnter);
+          assert.isFalse(Navigation.isCurrentPanel('thread-list'));
+          assert.isFalse(elements.InboxView.classList.contains('panel-active'));
+          assert.equal(elements.InboxView.getAttribute('aria-hidden'), 'true');
+          sinon.assert.notCalled(console.error);
+        }).then(done, done);
+      });
+
+      test('Clear pending when launched by notification and enter other panel',
+        function(done) {
+
+        navigator.mozHasPendingMessage.withArgs('notification').returns(true);
+
+        Navigation.init().catch(() => {}).then(() => {
+          assert.isTrue(Navigation.hasPendingInit());
+        }).then(() => Navigation.toPanel('composer')).then(() => {
```

Comment:
you can simply add `return Navigation.toPanel('composer')` in the previous then handler